### PR TITLE
Yanlam/filter

### PR DESCRIPTION
**Pull request by CameronHamidi**
## Overview

Adds a filter view controller accessible from the Sports tab for searching for pickup games.

## Changes Made

Added `SportsFilterViewController` and several collection view cell classes to represent gym selection, start time, number of players, and sports.

## Test Coverage

Verified visually that the app matches the designs.

## Next Steps (delete if not applicable)

Add in networking functionality.

## Screenshots (delete if not applicable)

See #uplift-ios for demonstration video

### CODE REVIEW

File: Uplift/Controllers/OnboardingLetterViewController.swift [hunk 1]
```diff
@@ -10,26 +10,25 @@ import UIKit
 import SnapKit
 
 class OnboardingLetterViewController: UIViewController {
-    
-    //MARK: - UI Elements + Init
+
+    //MARK:- UI Elements + Init
```
Comment (YKo20010) [inline Uplift/Controllers/OnboardingLetterViewController.swift:14]: space after //

File: Uplift/Controllers/SportsFilterViewController.swift [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        view.backgroundColor = .white
+
+        //NAVIGATION BAR
```
Comment (YKo20010) [inline Uplift/Controllers/SportsFilterViewController.swift:40]: space after //

File: Uplift/Views/Dropdown/DropdownHeaderView.swift [hunk 3]
```diff
@@ -75,49 +79,59 @@ class DropdownHeaderView: UIView {
...
+//        }
         isArrowRotated = true
     }
 
     func rotateArrowUp() {
-        UIView.animate(withDuration: 0.3) {
+//        UIView.animate(withDuration: 0.3) {
             self.arrowImageView.transform = CGAffineTransform(rotationAngle: 0)
-        }
+//        }
```
Comment (YKo20010) [inline Uplift/Views/Dropdown/DropdownHeaderView.swift:129]: If we don't need the UIView.animate lines, then just delete them.

File: Uplift/Models/SportsDropdown.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+//
+
+import Foundation
+import UIKit
+
+class DropdownData {
+
+    var completed: Bool!
+    var dropStatus: DropdownStatus!
+    var titles: [String]!
```
Comment (kevinchan159) [inline Uplift/Models/SportsDropdown.swift:16]: does this need to be a class? we could make it a struct and change all the `var` to `let` in which case you wouldn't need the init

File: Uplift/Models/SportsDropdown.swift [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        self.dropStatus = dropStatus
+        self.titles = titles
+    }
+
+}
+
+struct GymNameId: Equatable {
+
+    var name: String!
+    var id: String!
```
Comment (kevinchan159) [inline Uplift/Models/SportsDropdown.swift:29]: change to let

File: Uplift/Views/Cells/GymFilterCell.swift [hunk 6]
```diff
@@ -11,10 +11,12 @@ import UIKit
...
 
-    // MARK: - INITIALIZATION
-    static let identifier = Identifiers.gymFilterCell
-    var gymNameLabel: UILabel!
-    var selectedCircle: UIView!
+    static let labelFont = UIFont._14MontserratLight
+
+    var gymNameLabel = UILabel()
+    var rightDivider = UIView()
+    var selectedCircle = UIView()
```
Comment (kevinchan159) [inline Uplift/Views/Cells/GymFilterCell.swift:18]: change to private let

File: Uplift/Views/SportsFilter/SportsFilterSportsDropdownCollectionViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    private let halfViewHeight: CGFloat = 20
+    private let halfOpenCellNumber = 3
+
+    private var dropdownView: DropdownView!
+
+    weak var dropdownDelegate: DropdownViewDelegate?
+
+    var selectedSports: [String] = []
+
+    var sportsTypeDropdownData: DropdownData!
```
Comment (kevinchan159) [inline Uplift/Views/SportsFilter/SportsFilterSportsDropdownCollectionViewCell.swift:29]: private?

File: Uplift/Views/SportsFilter/SportsFilterNumPlayersCollectionViewCell.swift [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+class SportsFilterNumPlayersCollectionViewCell: SportsFilterCollectionViewCell, RangeSeekSliderDelegate {
+
+    static let height: CGFloat = 103
+
+    private let numPlayersLabel = UILabel()
+    private let numPlayersSlider = RangeSeekSlider(frame: .zero)
+
+    let maxPlayers = 10
+    let minPlayers = 2
```
Comment (kevinchan159) [inline Uplift/Views/SportsFilter/SportsFilterNumPlayersCollectionViewCell.swift:20]: private?

File: Uplift/Views/SportsFilter/SportsFilterStartTimeCollectionViewCell.swift [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+    static let height: CGFloat = 103
+
+    private let startTimeLabel = UILabel()
+    private let startTimeSlider = RangeSeekSlider(frame: .zero)
+
+    var endTime = "10:00PM"
+    var startTime = "6:00AM"
+    var timeFormatter = DateFormatter()
+    var timeRanges: [Date] = []
```
Comment (kevinchan159) [inline Uplift/Views/SportsFilter/SportsFilterStartTimeCollectionViewCell.swift:21]: private?

**Pull request merged**